pkg/metrics/metrics.go

```diff
@@ -0,0 +1,275 @@
+
+	views := []*view.View{
+		&view.View{
+			Description: AssignedIdentityAdditionDurationM.Description(),
+			Measure:     AssignedIdentityAdditionDurationM,
+			Aggregation: view.Distribution(0, 1, 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 97),
+		},
+		&view.View{
+			Description: AssignedIdentityAdditionCountM.Description(),
+			Measure:     AssignedIdentityAdditionCountM,
+			Aggregation: view.Count(),
+		},
+		&view.View{
+			Description: AssignedIdentityDeletionDurationM.Description(),
+			Measure:     AssignedIdentityDeletionDurationM,
+			Aggregation: view.Distribution(0, 1, 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 97),
+		},
+		&view.View{
+			Description: AssignedIdentityDeletionCountM.Description(),
+			Measure:     AssignedIdentityDeletionCountM,
+			Aggregation: view.Count(),
+		},
+		&view.View{
+			Description: NodeManagedIdentityOperationsDurationM.Description(),
+			Measure:     NodeManagedIdentityOperationsDurationM,
+			Aggregation: view.Distribution(0, 1, 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 97),
+			TagKeys:     []tag.Key{operationTypeKey, statusCodeKey},
+		},
+		&view.View{
+			Description: ManagedIdentityControllerCycleDurationM.Description(),
+			Measure:     ManagedIdentityControllerCycleDurationM,
+			Aggregation: view.Distribution(0, 1, 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 97),
+		},
+		&view.View{
+			Description: ManagedIdentityControllerCycleCountM.Description(),
+			Measure:     ManagedIdentityControllerCycleCountM,
+			Aggregation: view.Count(),
+		},
+		&view.View{
+			Description: ManagedIdentityControllerNewLeaderElectionCountM.Description(),
+			Measure:     ManagedIdentityControllerNewLeaderElectionCountM,
+			Aggregation: view.Count(),
+		},
+		&view.View{
+			Description: CloudProviderOperationsErrorsCountM.Description(),
+			Measure:     CloudProviderOperationsErrorsCountM,
+			Aggregation: view.Count(),
+			TagKeys:     []tag.Key{operationTypeKey},
+		},
+		&view.View{
+			Description: KubernetesAPIOperationsErrorsCountM.Description(),
+			Measure:     KubernetesAPIOperationsErrorsCountM,
+			Aggregation: view.Count(),
+			TagKeys:     []tag.Key{operationTypeKey},
+		},
+	}
+
+	err := view.Register(views...)
+	return err
+}
+
+// record records the given measure
+func record(ctx context.Context, ms ...stats.Measurement) {
+	stats.Record(ctx, ms...)
+}
+
+// reporter is stats reporter in the context
+type reporter struct {
+	ctx context.Context
+}
+
+// Report records the given measure
+func (r *reporter) Report(ms ...stats.Measurement) {
+	record(r.ctx, ms...)
+}
+
+// NewReporter creates a reporter with new context
+func NewReporter() (*reporter, error) {
+	ctx, err := tag.New(
+		context.Background(),
+	)
+	if err != nil {
+		return nil, err
+	}
+	return &reporter{ctx: ctx}, nil
+}
+
+// NewReporterAndReport creates a new instance of reporter and report given measurements
+func NewReporterAndReport(ms ...stats.Measurement) {
```

Comment:
If we add the reporter into the structs where we report, I don't think we will need this function.

Reply:
Agree,removed.